client/extensions/woocommerce/app/settings/email/mailchimp/setup-mailchimp.js

```diff
@@ -0,0 +1,315 @@
+	}
+
+	onKeyInputChange = ( e ) => {
+		this.setState( { api_key_input: e.target.value } );
+	}
+
+	// Right now Store info is combination of values from SettingsPaymentsLocationCurrency
+	// and managed directly - not the greatest option but good for now.
+	onStoreInfoChange = ( e ) => {
+		this.setState( { settings: Object.assign( {}, this.state.settings, { [ e.target.name ]: e.target.value } ) } );
+	}
+
+	renderStep = () => {
+		const { step } = this.state;
+		if ( step === LOG_INTO_MAILCHIMP_STEP ) {
+			return <LogIntoMailchimp />;
+		}
+		if ( step === KEY_INPUT_STEP ) {
+			return <KeyInputStep
+				onChange={ this.onKeyInputChange }
+				apiKey={ this.state.api_key_input }
+				isKeyCorrect={ this.props.isKeyCorrect } />;
+		}
+		if ( step === STORE_INFO_STEP ) {
+			return <StoreInfoStep
+				onChange={ this.onStoreInfoChange }
+				storeData={ this.state.settings }
+				validateFields={ false }
+			/>;
+		}
+		if ( step === CAMPAIGN_DEFAULTS_STEP ) {
+			return <CampaignDefaultsStep
+				onChange={ this.onStoreInfoChange }
+				storeData={ this.state.settings }
+				validateFields={ false }
+			/>;
+		}
+		if ( step === NEWSLETTER_SETTINGS_STEP ) {
+			return <NewsletterSettings
+				onChange={ this.onStoreInfoChange }
+				storeData={ this.state.settings }
+				validateFields={ false }
+			/>;
+		}
+		return <div></div>;
+	}
+
+	render() {
+		const { translate, hasMailChimp, siteId } = this.props;
+		const isButtonBusy = this.props.isBusy ? 'is-busy' : '';
+		const buttons = [
+			{ action: 'cancel', label: translate( 'Cancel' ) },
+			{ action: 'next', label: translate( 'Next' ), onClick: this.next, isPrimary: true, additionalClassNames: isButtonBusy },
+		];
+
+		const dialogClass = 'woocommerce mailchimp__setup';
+		const stepNum = steps[ this.state.step ].number;
+		if ( ! hasMailChimp ) {
+			return (
+				<Dialog
+					isVisible={ true }
+					buttons={ null }
+					className={ dialogClass }>
+					<div className="mailchimp__setup-dialog-title">MailChimp</div>
+					<RequiredPluginsInstallView
+						site={ { ID: siteId } }
+						skipConfirmation ={ true } />
+				</Dialog>
+			);
+		}
+		return (
+				<Dialog
+					isVisible={ true }
+					buttons={ buttons }
+					onClose={ this.onClose }
+					className={ dialogClass }>
+					<div className="mailchimp__setup-dialog-title">MailChimp</div>
+					<ProgressBar
+						value={ stepNum + 1 }
+						total={ 5 }
+						compact={ true } />
+					<ProgressIndicator
+						stepNumber={ stepNum }
+						totalSteps={ 5 } />
+						<div className="mailchimp__setup-dialog-content">
+							{ this.renderStep() }
+						</div>
+				</Dialog>
+		);
+	}
+}
+
+export default localize( connect(
+	( state, props ) => {
+		const subbmittingApiKey = isSubbmittingApiKey( state, props.siteId );
+		const isKeyCorrect = isApiKeyCorrect( state, props.siteId );
+		const address = getStoreLocation( state );
+		const currency = getCurrencyWithEdits( state );
+		const isBusy = subbmittingApiKey;
```

Comment:
another double b submit

Reply:
I will deal with the double b in the final PR - I don't want to add more files to this PR - for clarity.